### DISCUSSION TITLE
FR: Simplifying Dark Detection in Magic Areas with Illuminance Sensors

### DISCUSSION THREAD

**Discussion by Julien-Decoen** [jseidl/magic-areas#416] [Ideas / Feature Requests]
Hi @jseidl ,

I’d like to discuss how Magic Areas defines whether a room is dark or not. Based on what I understand, here’s the process if I want to use my illuminance sensors to determine if a room is dark:

1/ Add illuminance sensors in my Home Assistant area.
2/ Enable the sensor aggregation feature.
3/ Adjust the Illuminance threshold value.
4/ Select the newly created Threshold Light binary_sensor in the "Advanced area state tracking" section.

If I got this process right, what do you think about making the Threshold Light binary_sensor the default for managing room lighting detection? It could simplify this use case by making it the standard setup for handling room illuminance.

Looking forward to your thoughts!

Julien

**Comment by jseidl** [comment]
Hey @Julien-Decoen, the reason I don't do that is because its tricky to use the rooms own illuminance as light sensor without finagling with the hysteresis. It will cause an eternal loop of lights going on and off reacting to the illuminance level.

I use the exterior light sensor in almost all of the rooms. So steps 1-3 on your workflow would be done in the Exterior area and 4 on each area.

**Reply by Julien-Decoen** [reply]
> Hey @Julien-Decoen, the reason I don't do that is because its tricky to use the rooms own illuminance as light sensor without finagling with the hysteresis. It will cause an eternal loop of lights going on and off reacting to the illuminance level.
> 
> I use the exterior light sensor in almost all of the rooms. So steps 1-3 on your workflow would be done in the Exterior area and 4 on each area.

Thanks for the explanation! I’m wondering if the same issue with the illuminance and hysteresis happens when using sun.sun as a light source instead. Could using the sun.sun sensor help avoid the loop you described? If it works better, maybe we could even set it as the default for handling lighting control in rooms.

**Comment by Julien-Decoen** [comment]
I hadn't thought about using exterior light sensors to adapt the room lighting. That actually makes a lot of sense, although I'm still trying to figure out if there could be any side effects.

On my end, I’ve modified Magic Areas to work like this: I have a light sensor in each room (or shared with another room that has similar exposure using include_entities). For configuration, I use several entities per room:

- A number entity with the lux level at which the room's light is at maximum brightness.
- A number entity with the minimum brightness for the room.
- A number entity with the maximum brightness for the room.
- A select entity with options (small = 10 lux / medium = 25 lux / large = 50 lux / extra large = 100 lux) to define the range over which the light adjusts gradually. The brighter the room, the larger the range, and vice versa.
<br />
<img width="339" alt="image" src="https://github.com/user-attachments/assets/a16c0517-f6c8-47c1-b8e6-51181b51fcc5">
<br />
Here’s what the settings popup looks like for each room.
<br />
<img width="592" alt="image" src="https://github.com/user-attachments/assets/ba14c36f-b4ce-4843-a327-ba5379411b22">
<br />
<br />

To calculate the desired light level, I’ve used the formula from this blueprint: [Smart Lux Dimmer - Adjust Light Brightness Depending on Light Sensor Value](https://community.home-assistant.io/t/smart-lux-dimmer-adjust-light-brightness-depending-on-light-sensor-value/403646).

This implementation is quite advanced. I’m not sure if it aligns with your vision for Magic Areas, but I’m sharing it because I feel your implementation of the Threshold Light binary sensor is starting to resemble mine. Maybe we can share ideas and collaborate on this!

So far, the main limitation I’ve encountered is what I call the "cloud syndrome." When a cloud passes in front of my window, it causes the light to brighten up, and then a few seconds later, when the cloud moves away, the brightness drops back down abruptly. I’ve thought about fixing this by using an average of the aggregated illuminance over a set period, but I haven’t had time to implement it yet.

Let me know what you think. I believe using light sensors really adds a lot to Magic Areas!

**Comment by jseidl** [comment]
Hey @Julien-Decoen I intentionally left that out of MA in favor of using https://github.com/basnijholt/adaptive-lighting/, give it look

**Reply by Julien-Decoen** [reply]
Hi @jseidl ! In my current setup, I’m using adaptive-lighting with lux sensors to dynamically adjust the brightness of my lights. As the lux levels decrease in the room, the brightness of the lights increases accordingly. This allows for a smooth and intuitive lighting experience based on actual ambient conditions. I understand that this approach might not fully align with your vision for Magic Areas. :)
